src/cds_objects.cc

```diff
@@ -101,11 +101,8 @@ int CdsObject::resourcesEqual(const std::shared_ptr<CdsObject>& obj)
         return 0;
 
     // compare all resources
-    for (size_t i = 0; i < resources.size(); i++) {
-        if (!resources[i]->equals(obj->resources[i]))
-            return 0;
-    }
-    return 1;
+    return std::all_of(resources.begin(), resources.end(),
```

Comment:
Looks fishy: isn't that r always a one past last element in obj->resources?

Original loop indexed elements of both vectors, while with this change r is fixed, ie captured at lambda creation.

`std::equal` with custom predicate is better algorithm in this case.

Reply:
Correct. I took the code from your earlier comment.

Reply:
Oh, I found the one you training about ☺️

Looks like I didn't pay enough attention to the change add a whole and suggested only mechanical transformation of your change😞 Happens when reviewing on a phone (where screen is tiny and no access to all code...).

Reply:
Hmmm I'm not sure about this change. Converting this to a range loop gives

```
    for (auto res : resources) {
        if (!res->equals(obj->resources[std::distance(resources.begin(), resources.end())]))
            return 0;
    }
```

That's quite verbose.

Reply:
Meh reverted.